File: laboratories/ppo/exercises/lab09/php/src/Handler.php
<?php

namespace Ppo\Lab09;

use Ppo\Lab09\Exceptions\InputException;
use Throwable;

class Handler
{
    protected ?Throwable $exception = null;

    public function handle(Throwable $exception): static
    {
        $this->exception = $exception;

        echo $exception->getMessage() . PHP_EOL;

        return $this;
    }

    public function shouldBreak(): bool
    {
        return $this->exception instanceof InputException && $this->exception->shouldBreak();
    }
}

File: laboratories/ppo/exercises/lab09/python/src/Handler.py
from src.exceptions.InputException import InputException


class Handler:
    def __init__(self):
        self.exception = None

    def handle(self, exception):
        self.exception = exception
        print(exception)
        return self

    def should_break(self):
        return isinstance(self.exception, InputException) and self.exception.should_break
